yb-voyager/test/utils/voyager_cmd_runner.go

```diff
@@ -0,0 +1,186 @@
+package testutils
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"os"
+	"os/exec"
+	"strconv"
+	"strings"
+	"time"
+
+	log "github.com/sirupsen/logrus"
+	testcontainers "github.com/yugabyte/yb-voyager/yb-voyager/test/containers"
+)
+
+type ExitCode int
+
+const (
+	ExitCodeSuccess ExitCode = 0
+	ExitCodeFailure ExitCode = 1
+)
+
+func (ec ExitCode) String() string {
+	switch ec {
+	case ExitCodeSuccess:
+		return "Success"
+	case ExitCodeFailure:
+		return "Failure"
+	default:
+		return fmt.Sprintf("Exitcode(%d)", int(ec))
+	}
+}
+
+// wrapper for exec.Cmd to capture stdout and stderr and to extend it as per our test framework needs
+type VoyagerCommandRunner struct {
+	// basic command information to be provided
+	container   testcontainers.TestContainer
+	CmdName     string
+	CmdArgs     []string
+	isAsync     bool
+	doDuringCmd func()
+
+	// as per test framework needs
+	Cmd       *exec.Cmd
+	StdoutBuf *bytes.Buffer
+	StderrBuf *bytes.Buffer
+	exitCode  ExitCode
+	lastErr   error // lastErr is used to capture the last error encountered during command execution
+}
+
+func NewVoyagerCommandRunner(container testcontainers.TestContainer, cmdName string, cmdArgs []string, doDuringCmd func(), isAsync bool) *VoyagerCommandRunner {
+	if cmdName == "" {
+		log.Fatal("Command name cannot be empty")
+	}
+
+	if container == nil && (isSourceCmd(cmdName) || isTargetCmd(cmdName)) {
+		log.Fatal("Container cannot be nil for source/target commands")
+	}
+
+	cmdRunner := VoyagerCommandRunner{
+		container:   container,
+		CmdName:     cmdName,
+		CmdArgs:     cmdArgs,
+		doDuringCmd: doDuringCmd,
+		isAsync:     isAsync,
+		exitCode:    ExitCodeSuccess,
+	}
+	log.Debugf("Creating CommandRunner for command: %s with args: %s", cmdName, strings.Join(cmdArgs, " "))
+	return &cmdRunner
+}
+
+func (this *VoyagerCommandRunner) Prepare() error {
```

Comment:
`this` is generally not golang idiomatic :) https://go.dev/wiki/CodeReviewComments#receiver-names

I'm curious as to what made you choose "this", since it's more a java thing 😁

Reply:
> what made you choose "this"

One of two hard problems in CS 😅 - https://martinfowler.com/bliki/TwoHardThings.html

Was not able to comeup with a convincing var name at that moment